### PR TITLE
tests: Replaces Redis image with Agnhost

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind feature

/sig testing
/sig windows
/area conformance

**What this PR does / why we need it**:

Recently, the version of redis used in Kubernetes E2E tests has been bumped to 5.0.5 (for IPv6 support), but the latest supported version for Windows is 3.2. This can lead to failing tests, the output and behaviour can be different (see https://github.com/kubernetes/kubernetes/pull/80516). In order to prevent such failures, the amount of times the Redis image is used can be reduced..

Some tests are using the Redis image, but they do not explicitly need it. This PR replaces the usage of the Redis image with the Agnhost image in such test cases.

**Which issue(s) this PR fixes**:

Fixes #TBA

**Special notes for your reviewer**:

Affected tests:
    # gce only. doesn't seem to matter what image is used, it only tests / checks the deployment itself.
    should audit API calls to create, get, update, patch, delete, list, watch deployments.

    # daemonset image from httpd to redis.
    should not update pod when spec was updated and update strategy is OnDelete
    should update pod when spec was updated and update strategy is RollingUpdate

    # creates webserver replicaset, create deployment with non-existent image, rollover old replicaset for deployment with new image (redis)
    deployment should support rollover

    # creates webserver replicaset, creates redis deployment
    RollingUpdateDeployment should delete old pods and create new ones

    # creates redis deployment, waits for it to finish, updates it with webserver image
    RecreateDeployment should delete old pods and create new ones

    # creates webserver replcaset, creates deployment with agnhost image (with revision history limit = 0)
    deployment should delete old replica sets

    # creates a replication controller, applies some changes (new label, selector)
    should apply a new configuration to an existing RC

    # creates service, applies the same service, ensures node port stays the same.
    should reuse port when apply to an existing SVC

    # creates RC and service and checks kubectl describe outputs.
    should check if kubectl describe prints relevant information for rc and pods

    # creates RC controller, waits for pod to start, looks for specific string in pod logs.
    should create services for rc 

    # creates RC, patches a new annotation.
    should add annotations for pods in rc

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/approve
this lgtm from my end, but since this was discussed during sig-apps I'll let the final word to @kow3ns 
/assign @kow3ns

File: test/e2e/testing-manifests/kubectl/redis-master-pod.yaml [hunk 1]
```diff
@@ -1,33 +0,0 @@
-apiVersion: v1
```
Comment (liggitt) [inline test/e2e/testing-manifests/kubectl/redis-master-pod.yaml]: test-cmd was using this file. failures were masked by https://github.com/kubernetes/kubernetes/commit/313044abd7a23699b5d744c4157ca7392673d587#diff-7976a1974df5995ca28f32cc945e92acL135. fixing in https://github.com/kubernetes/kubernetes/pull/84871

**Comment by bclau** [comment]
@listx the agnhost image needs to be rebuilt.

**Pull request merged**